### PR TITLE
Operate First - Minimizing RBAC Requirements

### PR DESCRIPTION
**Pull request by eletonia**
This PR adds namespaced roles in the Helm templates instead of a cluster role to minimize RBAC requirements to deploy M4D on Operate First

### CODE REVIEW

**Comment by eletonia** [comment]
@roee88 I changed the `m4d-blueprints` role to have access to all resources, added role bindings for the `m4d-system` and `m4d-blueprints` roles, and added cluster roles for `m4dapplication`, `batchtransfer`, and `streamtransfer` Can you review these changes?

**Comment by eletonia** [comment]
The `m4d-blueprints` and `m4d-system` roles and role bindings now check for `coordinator` and the m4dapplication cluster role now checks for `worker` @roee88 Does this look correct?

File: charts/m4d/templates/m4dapplication-cr.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{- if include "m4d.isEnabled" (tuple .Values.manager.enabled (and .Values.worker.enabled)) }}
+{{- if .Values.clusterScoped }}
+# ClusterRole m4dapplication-editor allows managing m4dapplications
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: m4dapplication-editor
```
Comment (roee88) [inline charts/m4d/templates/m4dapplication-cr.yaml:7]: Consider renaming to `m4d-user` or something similar to reflect that it's a role for users of M4D

**Review by roee88 — CHANGES_REQUESTED** [review]

File: charts/m4d/templates/m4d-system-role.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  namespace: m4d-system
+  name: {{ template "m4d.fullname" . }}-system-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - plotters
+  - m4dstorageaccounts
+  - m4dmodules
+  - datasets
```
[the remark this answers is not in the document]
Comment (eletonia) [reply charts/m4d/templates/m4d-system-role.yaml:13]: I checked `manager-cr.yaml` and only plotters has a `/status` so I just added that to this file

**Comment by eletonia** [comment]
Thanks for reviewing @roee88 
I made the suggested changes and added the new `m4d-admin` cluster role
Can we merge?

**Review by roee88 — APPROVED** [review]

**Comment by roee88** [comment]
> Thanks for reviewing @roee88
> I made the suggested changes and added the new `m4d-admin` cluster role
> Can we merge?

Approved. I'll check it manually next week since the CI is still broken and then merge.

**Comment by ronenkat** [comment]
@eletonia can you rebase? let see if we pass all tests

File: charts/m4d/templates/m4d-blueprints-rb.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{{- if include "m4d.isEnabled" (tuple .Values.manager.enabled .Values.worker.enabled) }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ template "m4d.fullname" . }}-blueprints-rb
```
Comment (roee88) [inline charts/m4d/templates/m4d-blueprints-rb.yaml:5]: Missing namespace, should be set to `m4d-blueprints`. Thanks @shlomitk1.

**Review by roee88 — CHANGES_REQUESTED** [review]

**Comment by eletonia** [comment]
@ronenkat @roee88 I rebased and added the `m4d-blueprints` namespace to the m4d-blueprints role binding.
Now the integration tests are failing for batch and stream transfers. The `streamtransfer-sample` secret and `batchtransfer-sample` job are not being created for some reason, do you know why?

**Comment by shlomitk1** [comment]
> 
> 
> @ronenkat @roee88 I rebased and added the `m4d-blueprints` namespace to the m4d-blueprints role binding.
> Now the integration tests are failing for batch and stream transfers. The `streamtransfer-sample` secret and `batchtransfer-sample` job are not being created for some reason, do you know why?

The yamls in manager/testdata (batchtransfer and streamtransfer) should be invoked in m4d-blueprints rather than in default namespace.

**Comment by eletonia** [comment]
> > @ronenkat @roee88 I rebased and added the `m4d-blueprints` namespace to the m4d-blueprints role binding.
> > Now the integration tests are failing for batch and stream transfers. The `streamtransfer-sample` secret and `batchtransfer-sample` job are not being created for some reason, do you know why?
> 
> The yamls in manager/testdata (batchtransfer and streamtransfer) should be invoked in m4d-blueprints rather than in default namespace.

I changed the namespace in [batchtransfer.yaml](https://github.com/eletonia/the-mesh-for-data/blob/operate-first-rbac/manager/testdata/batchtransfer.yaml) and [streamtransfer.yaml](https://github.com/eletonia/the-mesh-for-data/blob/operate-first-rbac/manager/testdata/streamtransfer.yaml) to `m4d-blueprints` but now the unit tests are failing since it can't find the `m4d-blueprints` namespace:
```
Should simulate a StreamTranssfer successfully [It]
    /home/runner/work/mesh-for-data/mesh-for-data/manager/controllers/motion/streamtransfer_controller_test.go:47

    Expected success, but got an error:
        <*errors.StatusError | 0xc0003e2a00>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "namespaces \"m4d-blueprints\" not found",
                Reason: "NotFound"
```

**Comment by roee88** [comment]
@shlomitk1 take a look at https://github.com/kubernetes-sigs/controller-runtime/pull/1435.

**Comment by shlomitk1** [comment]
> 
> 
> @shlomitk1 take a look at [kubernetes-sigs/controller-runtime#1435](https://github.com/kubernetes-sigs/controller-runtime/pull/1435).

It is supported starting from v0.9.0. This breaks dependencies, I've tried to upgrade other k8s dependencies but it causes problems with required dependencies for datashim...

**Review by roee88 — APPROVED** [review]

**Pull request merged**